action/sealedenvelope.go

```diff
@@ -14,44 +14,46 @@ import (
 // SealedEnvelope is a signed action envelope.
 type SealedEnvelope struct {
 	Envelope
-	encoding      iotextypes.Encoding
-	externChainID uint32
-	srcPubkey     crypto.PublicKey
-	signature     []byte
+	encoding     iotextypes.Encoding
+	evmNetworkID uint32
+	srcPubkey    crypto.PublicKey
+	signature    []byte
 }
 
-// rawHash returns the raw hash of embedded Envelope (this is the hash to be signed)
+// envelopeHash returns the raw hash of embedded Envelope (this is the hash to be signed)
 // an all-0 return value means the transaction is invalid
-func (sealed *SealedEnvelope) rawHash() hash.Hash256 {
-	if IsRLP(sealed.encoding) {
+func (sealed *SealedEnvelope) envelopeHash() (hash.Hash256, error) {
+	switch sealed.encoding {
+	case iotextypes.Encoding_ETHEREUM_RLP:
 		tx, err := actionToRLP(sealed.Action())
 		if err != nil {
-			return hash.ZeroHash256
+			return hash.ZeroHash256, err
 		}
-		h, err := rlpRawHash(tx, sealed.externChainID)
-		if err != nil {
-			return hash.ZeroHash256
-		}
-		return h
+		return rlpRawHash(tx, sealed.evmNetworkID)
+	case iotextypes.Encoding_IOTEX_PROTOBUF:
+		return hash.Hash256b(byteutil.Must(proto.Marshal(sealed.Envelope.Proto()))), nil
 	}
-	return sealed.Envelope.Hash()
+	return hash.ZeroHash256, errors.Errorf("unknown encoding type %s", sealed.encoding)
 }
 
 // Hash returns the hash value of SealedEnvelope.
 // an all-0 return value means the transaction is invalid
 func (sealed *SealedEnvelope) Hash() hash.Hash256 {
-	if IsRLP(sealed.encoding) {
+	switch sealed.encoding {
+	case iotextypes.Encoding_ETHEREUM_RLP:
 		tx, err := actionToRLP(sealed.Action())
 		if err != nil {
-			return hash.ZeroHash256
+			panic(err)
 		}
-		h, err := rlpSignedHash(tx, sealed.externChainID, sealed.Signature())
+		h, err := rlpSignedHash(tx, sealed.evmNetworkID, sealed.Signature())
 		if err != nil {
-			return hash.ZeroHash256
+			panic(err)
 		}
 		return h
+	case iotextypes.Encoding_IOTEX_PROTOBUF:
+		return hash.Hash256b(byteutil.Must(proto.Marshal(sealed.Proto())))
 	}
-	return hash.Hash256b(byteutil.Must(proto.Marshal(sealed.Proto())))
+	panic("unknown encoding type")
```

Comment:
if someone send in an action with valid ActionCore but encoding = 2, this would cause the node to panic and stop running

you can think 0-hash is a "hidden" knowledge, but IMO that a way to fail gracefully, and it is fine given the behavior is clearly documented and understood

Reply:
I thought your code https://github.com/iotexproject/iotex-core/pull/2601/files#diff-fda8afdb7165ec30b8d2c40dfd1963def0ffeb117a3d9098167b8139c72527caL105 covers the test

Reply:
no, that is to check, if encoding = 1, the ActionCore inside MUST be either Transfer or Execution

someone can send in a valid Transfer/Execution with encoding = 2, causing a panic

Reply:
how could that happen with https://github.com/iotexproject/iotex-core/pull/2601/files#diff-fda8afdb7165ec30b8d2c40dfd1963def0ffeb117a3d9098167b8139c72527caL96?

Reply:
ok, cannot get here
what about the `rlpSignedHash` above, in case err happens, it will panic
i think we better stay on safe side